src/components/PeoplePage.tsx

```diff
@@ -1,35 +1,116 @@
+import { useEffect, useState } from 'react';
+import { useSearchParams } from 'react-router-dom';
 import { PeopleFilters } from './PeopleFilters';
 import { Loader } from './Loader';
 import { PeopleTable } from './PeopleTable';
+import { Person } from '../types';
+import { getPeople } from '../api';
+import { getParents } from '../utils/getParents';
 
 export const PeoplePage = () => {
-  return (
-    <>
-      <h1 className="title">People Page</h1>
+  const [people, setPeople] = useState<Person[]>([]);
+  const [isLoading, setLoading] = useState(false);
+  const [hasError, setError] = useState(false);
+  const [searchParams] = useSearchParams();
 
-      <div className="block">
-        <div className="columns is-desktop is-flex-direction-row-reverse">
-          <div className="column is-7-tablet is-narrow-desktop">
-            <PeopleFilters />
-          </div>
+  const centuries = searchParams.getAll('centuries');
+  const query = searchParams.get('query') || '';
+  const sex = searchParams.get('sex') || '';
+
+  let visibleElement = null;
+
+  const filterByQuery = (person: Person) => {
+    const queryCaseIns = RegExp(query, 'i');
+
+    return queryCaseIns.test(person.name)
+    || queryCaseIns.test(person.motherName || '')
+    || queryCaseIns.test(person.fatherName || '');
+  };
+
+  const filterByCentury = (person: Person) => (
+    centuries.length
+      ? centuries.includes(Math.ceil(person.born / 100).toString())
+      : true
+  );
+
+  const filterBySex = (person: Person) => (
+    sex ? person.sex === sex : true
+  );
 
-          <div className="column">
-            <div className="box table-container">
-              <Loader />
+  const visiblePeople = people.filter(person => (
+    filterBySex(person)
+    && filterByQuery(person)
+    && filterByCentury(person)
+  ));
 
-              <p data-cy="peopleLoadingError">Something went wrong</p>
+  const fetchPeople = () => {
+    setLoading(true);
+    getPeople()
+      .then(response => {
+        const peopleWithParents = response.map(person => ({
+          ...person,
+          ...getParents(person, response),
+        }));
 
-              <p data-cy="noPeopleMessage">
-                There are no people on the server
-              </p>
+        setPeople(peopleWithParents);
+      })
+      .catch(() => setError(true))
+      .finally(() => setLoading(false));
+  };
 
-              <p>There are no people matching the current search criteria</p>
+  useEffect(() => {
+    fetchPeople();
+  }, []);
 
-              <PeopleTable />
-            </div>
+  switch (true) {
```

Comment:
this is a bit weird `switch(true)`, just not a pattern you could often see
i get the idea,
yet i'd rather nest the conditional rendering logic inside return statement
```
const renderMainElement = !isLoading
  && !hasError
  && people.length

return (
  {isLoading && <Loader />}
  {hasError && ...Something went wrong}
  {!people.length && ...There are no people on the server}
  {renderMainElement && ...}
```